pkg/scheduler/apis/config/validation/validation_pluginargs.go

```diff
@@ -0,0 +1,133 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package validation
+
+import (
+	"fmt"
+
+	v1 "k8s.io/api/core/v1"
+	metav1validation "k8s.io/apimachinery/pkg/apis/meta/v1/validation"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	"k8s.io/kubernetes/pkg/scheduler/apis/config"
+)
+
+// ValidateInterPodAffinityArgs validates that InterPodAffinityArgs are correct.
+func ValidateInterPodAffinityArgs(args config.InterPodAffinityArgs) error {
+	return ValidateHardPodAffinityWeight(field.NewPath("hardPodAffinityWeight"), args.HardPodAffinityWeight)
+}
+
+// ValidateHardPodAffinityWeight validates that weight is within allowed range.
+func ValidateHardPodAffinityWeight(path *field.Path, w int32) error {
+	minHardPodAffinityWeight := int32(0)
```

Comment:
I'm not convinced of @ahg-g's suggestion here. This would allocate a variable each time the function is called.
Perhaps just doing `w < 0 || w > 100` is enough, but unexported constants would look cleaner.

Reply:
>  This would allocate a variable each time the function is called.

Pretty sure the compiler will optimize that away, even if it doesn't this function is called once on initialization. Regardless, hardcoding the values is fine as well, my point is that having a globally defined variable isn't justified.

Reply:
Both comments are valid I think. Declared as `const` inside the function which satisfies both arguments and clarifies the intent better.